tests/integration/mixer/policy/white_black_listing_test.go

```diff
@@ -33,45 +92,26 @@ func TestWhiteListing(t *testing.T) {
 			t.Fatalf("bookinfo namespace not allocated in setup")
 		}
 		bookinfoNs := *bookinfoNamespace
-		g.ApplyConfigOrFail(
-			t,
-			bookinfoNs,
-			bookinfo.NetworkingBookinfoGateway.LoadGatewayFileWithNamespaceOrFail(t, bookinfoNs.Name()))
-		g.ApplyConfigOrFail(
-			t,
-			bookinfoNs,
-			bookinfo.GetDestinationRuleConfigFile(t, ctx).LoadWithNamespaceOrFail(t, bookinfoNs.Name()),
-			bookinfo.NetworkingVirtualServiceAllV1.LoadWithNamespaceOrFail(t, bookinfoNs.Name()),
-		)
-		defer g.DeleteConfigOrFail(
-			t,
-			bookinfoNs,
-			bookinfo.NetworkingBookinfoGateway.LoadGatewayFileWithNamespaceOrFail(t, bookinfoNs.Name()),
-		)
-		defer g.DeleteConfigOrFail(
-			t,
-			bookinfoNs,
-			bookinfo.GetDestinationRuleConfigFile(t, ctx).LoadWithNamespaceOrFail(t, bookinfoNs.Name()),
-			bookinfo.NetworkingVirtualServiceAllV1.LoadWithNamespaceOrFail(t, bookinfoNs.Name()),
-		)
 
-		if ingInst == nil {
-			t.Fatalf("ingress not setup")
-		}
-		ing := *ingInst
-		// Verify you can access productpage right now.
-		util.SendTrafficAndWaitForExpectedStatus(ing, t, "Sending traffic...", "", 2, http.StatusOK)
+		g.ApplyConfigOrFail(t, bookinfoNs,
+			`apiVersion: v1
+kind: Service
+metadata:
+  name: productpage
+spec:
+  ports:
+  - port: 9080
+    name: tcp`)
+		defer g.ApplyConfigOrFail(t, bookinfoNs,
```

Comment:
is there not a `Setup` and `Teardown` hook in the framework? Having to restore config state via `kubectl apply` in a `defer` seems a bit odd.

also, would it be possible to have a second TCP service that redirects to the same port? seems like we might have to wire up the helper method to take a port arg... not sure how that works with the framework, as I don't see a URL get built anywhere in this method.

Reply:
Not that I'm aware of. The secondary service is an interesting idea, but if the real issue is with networking, then that won't solve the problem.

Reply:
there is a test suite wide setup but not specific test case specific... 
we can move common `kubectl apply` to testsetup fn ?